pyinteraph/path_analysis.py

```diff
@@ -0,0 +1,481 @@
+            # Find the index of those res in the indentifiers list
+            index = [identifiers.index(res) for res in colon_split]
+            # Replace with the residues in that range
+            inp_replace = identifiers[index[0]:index[1]+1]
+            # Concatenate to list
+            colon_replace += inp_replace
+        except Exception:
+            # Report if the specified range does not exist in the PDB
+            error_str = "Residue range not in PDB or incorrect format: {:s}"
+            raise ValueError(error_str.format(inp))
+    # Add both lists
+    input_list = comma_list + colon_replace
+    # Remove duplicates
+    input_list = list(set(input_list))
+    return input_list
+
+def get_shortest_paths(graph, source, target, maxl):
+    """Find all shortest paths between all combinations of source and
+    target.
+    """
+    
+    # Get all combinations
+    combinations = itertools.product(source, target)
+    # Get all shortest paths
+    paths = []
+    for node1, node2 in combinations:
+        if node1 != node2:
+            try:
+                # Get a list of shortest paths and append to output
+                path = list(nx.algorithms.shortest_paths.generic.all_shortest_paths(\
+                                G = graph, \
+                                source = node1, \
+                                target = node2))
+                for p in path:
+                    # Check that path is not longer than the maximum allowed length
+                    if len(p) <= maxl:
+                        paths.append(p)
+            except nx.NetworkXNoPath:
+                pass
+                # If no path is found log info
+                log.warning(f"No path found between {node1} and {node2}")
+    return paths
+
+def get_all_simple_paths(graph, source, target, maxl):
+    """Find all simple paths between all combinations of source and
+    target.
+    """
+
+    # Get all combinations
+    combinations = itertools.product(source, target)
+    # Get all simple paths
+    paths = []
+    for node1, node2 in combinations:
+        # Get all simple paths for each combination of source and target
+        path = list(nx.algorithms.simple_paths.all_simple_paths(\
+                        G = graph, \
+                        source = node1, \
+                        target = node2, \
+                        cutoff= maxl))
+        # Only add paths to output if they exist
+        for p in path:
+            if len(p) > 0:
+                paths.append(p)
+    return paths
+
+def sort_paths(graph, paths, sort_by):
+    """Takes in a list of paths and sorts them."""
+    
+    # Get source and target
+    source = [p[0] for p in paths]
+    target = [p[-1] for p in paths]
+    # Calculate length of path
+    lengths = [len(p) for p in paths]
+    #print(lengths)
+    # Calculate weights of path
+    weights = \
+    [[graph[p[i]][p[i+1]]["weight"] for i in range(len(p)-1)] for p in paths]
```

Comment:
Add an indentation to make it clearer. You can also split the list into multiple lines (i.e. at the second `for`) if it becomes too long.